### PR TITLE
Added null check post status obj in sal site class

### PR DESCRIPTION
**Pull request by darssen**
Fixes Warnings related to `class.json-api-site-base.php` like 
```
Attempt to read property "public" on null
Attempt to read property "private" on null
Attempt to read property "post_status" on null
Attempt to read property "protected" on null
```

## Proposed changes:

* Added null check since `get_post_status_object` can return null.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

Related logs f95ebb453440405af3826d85a96930ef-logstash
## Testing instructions:

Code Review

### CODE REVIEW

File: projects/plugins/jetpack/sal/class.json-api-site-base.php [hunk 1]
```diff
@@ -766,7 +766,7 @@ private function user_can_view_post( $post ) {
 			$post_status_obj = get_post_status_object( $post->post_status );
 		}
 
-		$authorized = (
+		$authorized = null !== $post_status_obj && (
```
Comment (chrisbliss18) [inline projects/plugins/jetpack/sal/class.json-api-site-base.php]: While this fixes the direct issue of the warning, it masks a logic error in the code above.

Attachments have a post status of `inherit`. It's possible for an attachment to not have a parent (an unattached attachment). So, the logic with the post status `inherit` check above needs to be changed to ensure that we don't check the parent if there isn't a parent. Perhaps logic such as the following:

```
if ( 'inherit' === $post->post_status && 0 !== (int)$post->post_parent ) {
```

Could you update the logic to ensure that we are checking the post status of the current post and not the parent in the event that a parent doesn't exist?

**Review by chrisbliss18 — CHANGES_REQUESTED** [review]
After talking with @fgiannar about your change, we realized that there is actually a logic problem that causes this warning. Rather than masking the warning with your check, we think that your check should be removed so that potential additional logic errors are not masked.

File: projects/plugins/jetpack/sal/class.json-api-site-base.php (hunk 1 above)
Comment (darssen) [reply projects/plugins/jetpack/sal/class.json-api-site-base.php]: Thanks for the review @chrisbliss18. This definitely makes sense. I've implemented it in [2bba21e](https://github.com/Automattic/jetpack/pull/39727/commits/2bba21efa85d237d92ffabb6ab8bb98510431cc8).
Let me know what you think

**Review by chrisbliss18 — APPROVED** [review]
The changes look good. They fix the logic error and don't mask potential future logic errors. Thanks for adding the comment explaining the reason for the added complexity in the check.

Approved. Thanks for the work on this. :)

**Pull request merged**